client/hwlib/src/collectors/cpuid.rs

```diff
@@ -0,0 +1,143 @@
+        if self.0.is_empty() {
+            return None;
+        }
+        let cpu_id_hex = format!("0x{:x}{:02x}{:02x}", self.0[2], self.0[1], self.0[0]);
+        Some(
+            self.to_human_friendly(&cpu_id_hex)
+                .unwrap_or("Unknown")
+                .to_string(),
+        )
+    }
+
+    /// Implement the same logic as used in checkbox to get CPU codename
+    /// https://github.com/canonical/checkbox/blob/904692d/providers/base/bin/cpuid.py
+    fn to_human_friendly(&self, cpu_id_hex: &str) -> Option<&'static str> {
+        let cpuid_map: &[(&str, &[&str])] = &[
+            ("Amber Lake", &["0x806e9"]),
+            ("AMD EPYC", &["0x800f12"]),
+            ("AMD Genoa", &["0xa10f11"]),
+            ("AMD Lisbon", &["0x100f81"]),
+            ("AMD Magny-Cours", &["0x100f91"]),
+            ("AMD Milan", &["0xa00f11"]),
+            ("AMD Milan-X", &["0xa00f12"]),
+            ("AMD ROME", &["0x830f10"]),
+            ("AMD Ryzen", &["0x810f81"]),
+            ("AMD Bergamo", &["0xaa0f01"]),
+            ("AMD Siena SP6", &["0xaa0f02"]),
+            ("AMD Raphael", &["0xa60f12"]),
+            ("Broadwell", &["0x4067", "0x306d4", "0x5066", "0x406f"]),
+            ("Canon Lake", &["0x6066"]),
+            ("Cascade Lake", &["0x50655", "0x50656", "0x50657"]),
+            (
+                "Coffee Lake",
+                &["0x806ea", "0x906ea", "0x906eb", "0x906ec", "0x906ed"],
+            ),
+            ("Comet Lake", &["0x806ec", "0xa065"]),
+            ("Cooper Lake", &["0x5065a", "0x5065b"]),
+            ("Emerald Rapids", &["0xc06f2"]),
+            ("Haswell", &["0x306c", "0x4065", "0x4066", "0x306f"]),
+            ("Hygon Dhyana Plus", &["0x900f22"]),
+            ("Ice Lake", &["0x606e6", "0x606a6", "0x706e6", "0x606c1"]),
+            ("Ivy Bridge", &["0x306a", "0x306e"]),
+            ("Kaby Lake", &["0x806e9", "0x906e9"]),
+            ("Knights Landing", &["0x5067"]),
+            ("Knights Mill", &["0x8065"]),
+            ("Nehalem", &["0x106a", "0x106e5", "0x206e"]),
+            ("Pineview", &["0x106ca"]),
+            ("Penryn", &["0x1067a"]),
+            ("Raptor Lake", &["0xb0671", "0xb06f2", "0xb06f5", "0xb06a2"]),
+            ("Rocket Lake", &["0xa0671"]),
+            ("Sandy Bridge", &["0x206a", "0x206d6", "0x206d7"]),
+            (
+                "Sapphire Rapids",
+                &["0x806f3", "0x806f6", "0x806f7", "0x806f8"],
+            ),
+            ("Skylake", &["0x406e3", "0x506e3", "0x50654", "0x50652"]),
+            ("Tiger Lake", &["0x806c1"]),
+            ("Alder Lake", &["0x906a4", "0x906A3", "0x90675", "0x90672"]),
+            ("Westmere", &["0x2065", "0x206c", "0x206f"]),
+            ("Whisky Lake", &["0x806eb", "0x806ec"]),
+        ];
+
+        for (name, ids) in cpuid_map {
+            for id in *ids {
+                if cpu_id_hex.contains(id) {
+                    return Some(name);
+                }
+            }
+        }
+        None
```

Comment:
We should use contains, not equality here, because for some codenames. You can check the `Knights Landing` as an example
This is the real CPU ID: `71 06 05 00 FF FB EB BF`
This is the pattern: `0x5067`

So, `cpu_id_hex` here is `0x50671`, and it should contain `0x5067`. If we use equality here, the logic won't work correctly

Reply:
I've replaced the loop with your approach